posts/2020-07-24-dev-mode-containers-preview.adoc

```diff
@@ -0,0 +1,74 @@
+---
+layout: post
+title: "Fast iterative development with Open Liberty dev mode and containers"
+categories: blog
+author_picture: https://avatars2.githubusercontent.com/u/29640130
+author_github: https://github.com/ericglau
+seo-title: Fast iterative development with Open Liberty dev mode and containers - OpenLiberty.io
+seo-description: Open Liberty's dev mode with container support allows you to iteratively develop cloud-native Java applications with your server running in a Docker container.
+blog_description: "Open Liberty's dev mode with container support allows you to iteratively develop cloud-native Java applications with your server running in a Docker container."
+---
+= Fast iterative development with Open Liberty dev mode and containers
+Eric Lau <https://github.com/ericglau>
+
+When developing an application that will eventually be brought to production, potential issues can be avoided by ensuring that your development and production environments are as similar as possible.  This aligns with link:/blog/2019/09/05/12-factor-microprofile-kubernetes.html[Twelve Factor App] methodology, particularly factor 10 which calls for dev/prod parity.  For cloud-native applications, part of this is addressed by using containers where your environment can be codified to give consistency in development and production.
+
+In this post, we introduce a technology preview for container support in Open Liberty dev mode.  This lets you develop applications on your local environment while your Open Liberty server runs in a container.  The development image is kept as similar as possible to the production image while allowing for iterative development, so that your code changes are automatically hot deployed to the container and picked up by the running server.  Additionally, dev mode lets you run tests either automatically or on demand, and you can attach a debugger at any time to debug your application.
+
+Note that since container support is in technology preview, its features and parameters are subject to change in future milestones or releases of the Liberty Maven or Gradle plugins.
```

Comment:
Can we add a call to action here i.e. invite the users to try it out and provide feedback?  And, embed a video demo?

Reply:
Added call to action.  Will add demo gifs in a separate commit.